api/python/quilt3/packages.py

```diff
@@ -1323,15 +1329,21 @@ def selector_fn(*args):
                     f"'build' instead."
                 )
 
-        if dest is None:
-            dest_parsed = registry_parsed.join(name)
+        if callable(dest):
+            def dest_fn(*args, **kwargs):
+                return registry_parsed.join(dest(*args, **kwargs))
```

Comment:
I'm not sure if I'm reading this right, but if the user provides a dest callable then we should give them control of the entire string, i.e. they might want to set the physical key to any registry/bucket. Here it looks like we restrict to registry= value?

Reply:
reworked it